src/Tizen.NUI.Components/Controls/ViewList.cs

```
@@ -0,0 +1,302 @@
+            /// Get the View object by the index.
+            /// </summary>
+            /// <param name="index"></param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public View this[int index]
+            {
+                get
+                {
+                    return internalViewList[index];
+                }
+                [EditorBrowsable(EditorBrowsableState.Never)]
+                set
+                {
+                }
+            }
+            /// <summary>
+            /// Get the index of a View.
+            /// </summary>
+            /// <param name="view">The View object to get the index of</param>
+            /// <returns>The index</returns>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public int IndexOf(View view)
+            {
+                return internalViewList.IndexOf(view);
+            }
+            /// <summary>
+            /// Insert a View in the list.
+            /// </summary>
+            /// <param name="order"></param>
+            /// <param name="view"></param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public void Insert(int order, View view)
+            {
+                if (order > internalViewList.Count)
+                {
+                    return;
+                }
+                internalViewList.Insert(order, view);
+                viewList.Insert(order, view);
+                if (viewList.selectedIndex == -1)
+                {
+                    viewList.Select(0);
+                }
+            }
+            /// <summary>
+            /// Remove a View from the list.
+            /// </summary>
+            /// <param name="order">The order of the to-be-removed object</param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public void RemoveAt(int order)
+            {
+                if (order > internalViewList.Count)
+                {
+                    return;
```

Please throw exception for unexpected behavior.